### PR TITLE
[OSDOCS-3609]: AWS machineset support for IMDS options

### PR DESCRIPTION
**Pull request by jeana-redhat**
Version(s):
4.11

Issue:
[OSDOCS-3609](https://issues.redhat.com//browse/OSDOCS-3609) for [OCPCLOUD-1436](https://issues.redhat.com//browse/OCPCLOUD-1436)

Link to docs preview:
- [Sample customized install-config.yaml file for AWS](http://file.rdu.redhat.com/jrouth/OSDOCS-3609-AWS-EC2-machineset-support-IMDSv2-IMDSv1//installing/installing_aws/installing-aws-customizations.html#installation-aws-config-yaml_installing-aws-customizations)
- [Machine set options for the Amazon EC2 Instance Metadata Service](http://file.rdu.redhat.com/jrouth/OSDOCS-3609-AWS-EC2-machineset-support-IMDSv2-IMDSv1//machine_management/creating_machinesets/creating-machineset-aws.html#machineset-imds-options_creating-machineset-aws)

Additional information:

### CODE REVIEW

File: modules/machineset-imds-options.adoc [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+You can use machine sets to create compute machines that use a specific version of the Amazon EC2 Instance Metadata Service (IMDS). Machine sets can create compute machines that allow the use of both IMDSv1 and link:https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html[IMDSv2], or compute machines that require the use of IMDSv2.
+
+To change the IMDS configuration for existing compute machines, edit the machine set YAML file that manages those machines. To create new compute machines with your preferred IMDS configuration, create a machine set YAML file with the appropriate values.
+
+The IMDS configuration for control plane machines is set during cluster installation. To change the control plane machine IMDS configuration, you must use the AWS CLI.
+
+[IMPORTANT]
+====
+Before configuring a machine set to create compute machines that require IMDSv2, ensure that your workloads support IMDSv2.
```
Comment (lobziik) [inline modules/machineset-imds-options.adoc:17]: >your workloads support IMDSv2.

I would be a bit more specific there, maybe:

>your workloads, which interacting with AWS metadata service are support IMDSv2.

Almost all workloads does not care about that IMDSes :)

Comment (lobziik) [inline modules/machineset-imds-options.adoc:17]: >your workloads support IMDSv2.

I would be a bit more specific there, maybe:

>your workloads, which interacting with AWS metadata service are support IMDSv2.

Almost all workloads does not care about that IMDSes :)

**Comment by lobziik** [comment]
LGTM :) Just one minor note about workloads warning

**Comment by lobziik** [comment]
LGTM :) Just one minor note about workloads warning

**Comment by lobziik** [comment]
/lgtm
Thanks!

**Comment by jeana-redhat** [comment]
@sunzhaohua2 this is ready for QA, PTAL

**Review by GroceryBoyJr — COMMENTED** [review]
/LGTM - stand by for merge

File: modules/machineset-creating-imds-options.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating_machinesets/creating-machineset-aws.adoc
+
+:_content-type: PROCEDURE
+[id="machineset-creating-imds-options_{context}"]
+= Configuring IMDS by using machine sets
+
+You can specify whether to require the use of IMDSv2 by adding or editing the value of `metadataServiceOptions.authentication` in your machine set YAML file.
```
Comment (kalexand-rh) [inline modules/machineset-creating-imds-options.adoc:9]: Maybe:
```suggestion
You can specify whether to require the use of IMDSv2 by adding or editing the value of `metadataServiceOptions.authentication` in the machine set YAML file for your compute machines.
```

Consider a mildly tangential note either here or in the concept with a link to how to do this with the AWS CLI for control plane machines.

**Review by kalexand-rh — COMMENTED** [review]
I have a few suggestions, but this is looking good!

File: modules/machineset-creating-imds-options.adoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+.Procedure
+* Add or edit the following line under the `providerSpec` field:
```
[the remark this answers is not in the document]
Comment (jeana-redhat) [reply modules/machineset-creating-imds-options.adoc:12]: Copy+paste got me :stuck_out_tongue:

File: modules/machineset-creating-imds-options.adoc (hunk 2 above)
Comment (jeana-redhat) [reply modules/machineset-creating-imds-options.adoc:9]: Great idea. Will add a link from where I talk about this in the concept as soon as I find one :slightly_smiling_face:

**Pull request merged**